apps/rewards/app/components/Content/MyRewards.js

```diff
@@ -14,48 +15,62 @@ import {
   ONE_TIME_DIVIDEND,
   RECURRING_DIVIDEND,
   ONE_TIME_MERIT,
+  READY,
+  CLAIMED,
 } from '../../utils/constants'
 import { Empty } from '../Card'
 import Metrics from './Metrics'
-import { useAppState } from '@aragon/api-react'
+import { useAppState, useAragonApi, useNetwork } from '@aragon/api-react'
 import BigNumber from 'bignumber.js'
-import { displayCurrency, getStatus } from '../../utils/helpers'
+import { displayCurrency, getStatus, getStatusId } from '../../utils/helpers'
 
 const MyRewards = ({
   myRewards,
   myMetrics,
   viewReward,
   claimReward,
+  claimHashes,
 }) => {
+  const network = useNetwork()
   const rewardsEmpty = myRewards.length === 0
+  const { api } = useAragonApi()
+  const [ user, setUser ] = useState()
+  api.accounts().subscribe(accounts => setUser(accounts[0]))
```

Comment:
Is this the same as `const { connectedAccount } = useAragonApi()`? See [docs here](https://hack.aragon.org/docs/api-js-ref-react#connectedaccount) – I think it might be.

Why include the user's address in the `claimHashes` structure? Why not add `transactionHash` to the `reward` directly, so we can get rid of this `user` logic, and simplify `getEtherescanLink` with something like:

```js
const link = `https://${networkChunk}etherscan.io/tx/${reward.transactionHash}`
```

Reply:
Because of the way rewards work. You create a reward, and then any token holder can claim it. Since multiple users should be able to claim a reward at any time, we need to track who claimed it. Your approach would only store the last claim, so every user would see the last claim made by any user, instead of their own claim.